lib/sync/transforms/branch.js

```diff
@@ -1,25 +1,54 @@
 const parseSmartCommit = require('../../transforms/smart-commit')
 const { getJiraId } = require('../../jira/util/id')
 
+/**
+ * mapBranch takes a branch node from the GraphQL response and
+ * attempts to find issueKeys in use anywhere in that object
+ *
+ * Locations can include:
+ *  - Branch Name (ref)
+ *  - Title of the associated Pull Request
+ *  - Messages from up to the last 100 commits in that branch
+ * @param {Object} branch
+ * @param {Object} repository
+ */
 function mapBranch (branch, repository) {
-  const { issueKeys } = parseSmartCommit(branch.associatedPullRequestTitle)
+  // setup an empty array to hold any issues keys we can find
+  const allKeys = []
 
-  if (!issueKeys) {
-    return
+  const { issueKeys: branchKeys } = parseSmartCommit(branch.name)
+  const { issueKeys: pullRequestKeys } = parseSmartCommit(branch.associatedPullRequestTitle)
+  const { issueKeys: commitKeys } = parseSmartCommit(branch.lastCommit.message)
+
+  if (branchKeys) {
+    allKeys.push(branchKeys)
+  }
+
+  if (pullRequestKeys) {
+    allKeys.push(pullRequestKeys)
   }
 
-  const { issueKeys: commitKeys } = parseSmartCommit(branch.lastCommit.message)
   if (commitKeys) {
-    issueKeys.push(commitKeys.filter(Boolean))
+    allKeys.push(commitKeys)
   }
 
-  const allKeys = issueKeys.filter(Boolean)
+  if (!allKeys) {
+    // If we get here, no issue keys were found anywhere they might be found
+    return
+  }
+
+  // If we do have some keys, let's flatten the array and filter outo any undefined values
+  const filteredKeys = allKeys.filter(Boolean)
```

Comment:
I really like the incremental implementation, that makes it very clear what is happening.
We *could* reduce the amount of conditionals by leaning into the array behavior a little bit.
Example:
```javascript
const allKeys = [].concat(branchKeys).concat(pullRequestKeys).concat(commitKeys)
const filteredKeys = allKeys.filter(boolean)

// or something like
const allKeys = [branchKeys, pullRequestKeys, commitKeys]
const filteredKeys = allKeys.filter(Boolean).flat() 

// The idea being that we dont need to care about the if checks since filter will do that for us:
> a.concat([]).concat([1,2,3]).concat(null).concat(undefined)
[ 1, 2, 3, null, undefined ]
> a.concat([]).concat([1,2,3]).concat(null).concat(undefined).filter(Boolean)
[ 1, 2, 3 ]
```

Reply:
Yeh I had actually intended to refactor this when I finished writing the tests, but I was getting tired. Implemented in 86a92becf76eb740de231c807db31418985447c3!